src/main/java/org/folio/oaipmh/helpers/AbstractGetRecordsHelper.java

```diff
@@ -160,11 +163,11 @@ private Response buildRecordsResponse(Context ctx, Request request,
           }
           // Some repositories like SRS can return record source data along with other info
           String source = storageHelper.getInstanceRecordSource(instance);
-          if (source != null) {
-            source = metadataManager.updateMetadataSourceWithDiscoverySuppressedDataIfNecessary(source, instance, request);
```

Comment:
In old implentation "discovery suppressed data" was processed even without record header's status checking. Now we process "suppredd discovery data" in couble wth rest of operation related to record filling if there is no header status to. Does it intended by business case?

Reply:
It is just reordering and optimizing ifs